tree/ntuple/v7/src/RPageStorageFile.cxx

```diff
@@ -313,8 +313,11 @@ void ROOT::Experimental::Internal::RPageSourceFile::LoadStructureImpl()
 {
    // If we constructed the page source with (ntuple name, path), we need to find the anchor first.
    // Otherwise, the page source was created by OpenFromAnchor()
-   if (!fAnchor)
+   if (!fAnchor) {
       fAnchor = fReader.GetNTuple(fNTupleName).Unwrap();
+   } else {
+      fReader.SetMaxKeySize(fAnchor->GetMaxKeySize());
+   }
```

Comment:
I am a bit confused.  Why is it not:
```suggestion
   if (!fAnchor) {
      fAnchor = fReader.GetNTuple(fNTupleName).Unwrap();
      fReader.SetMaxKeySize(fAnchor->GetMaxKeySize());
   }
```
and then the `SetMaxKeySize` also on line 305 ?

Reply:
The mini file reader will set the max key size as a side effect of reading the anchor.
If we already have the anchor, we don't need to read it again.

Reply:
@silverweed What do you think? I agree that the current situation is not ideal.

Reply:
I added @pcanal's suggestion to the PR